src/devices/src/virtio/vsock/event_handler.rs

```diff
@@ -427,31 +425,17 @@ mod tests {
         // - the backend doesn't have any pending RX data.
         {
             let test_ctx = TestContext::new();
-            let mut ctx = test_ctx.create_epoll_handler_context();
+            let mut ctx = test_ctx.create_event_handler_context();
 
-            ctx.handler.backend.set_pending_rx(false);
-            ctx.handler
-                .handle_event(BACKEND_EVENT, EventSet::IN)
-                .unwrap();
+            ctx.device.backend.set_pending_rx(false);
+            ctx.device.notify_backend(EpollEvent::new(EventSet::IN, 0));
 
             // The backend should've received this event.
-            assert_eq!(ctx.handler.backend.evset, Some(EventSet::IN));
+            assert_eq!(ctx.device.backend.evset, Some(EventSet::IN));
             // TX queue processing should've been triggered.
             assert_eq!(ctx.guest_txvq.used.idx.get(), 1);
             // The RX queue should've been left untouched.
             assert_eq!(ctx.guest_rxvq.used.idx.get(), 0);
         }
     }
-
-    #[test]
```

Comment:
Why not test this usecase?

Reply:
This is harder to effectively test right now, as `process` only logs a warning when an unexpected event is encountered.